In [1]:
import pandas as pd
import sqlite3

In [2]:
airbnb = pd.read_csv("../datasets/processed/new_york_cleaned.csv")

In [3]:
conn = sqlite3.connect("../sql/airbnb_new_york.db")

In [4]:
airbnb.to_sql(
    "airbnb_listings",
    conn,
    if_exists="replace",
    index=False
)

30259

In [5]:
query = """
SELECT *
FROM airbnb_listings
LIMIT 5;
"""

pd.read_sql(query, conn)

,id,neighbourhood_cleansed,latitude,longitude,property_type,room_type,accommodates,bedrooms,beds,price,availability_30,availability_60,availability_90,availability_365,number_of_reviews,review_scores_rating,reviews_per_month,estimated_occupancy_l365d,estimated_revenue_l365d,host_is_superhost
0,2539,Kensington,40.645937,-73.972164,Private room in condo,Private room,2,1.0,2.0,113.97,4,34,64,339,10,4.80,0.08,60,6838.0,No
1,6848,Williamsburg,40.709350,-73.953420,Entire rental unit,Entire home/apt,3,2.0,1.0,117.27,1,1,15,191,198,4.60,0.95,180,21109.0,Yes
2,6872,East Harlem,40.801070,-73.942550,Private room in condo,Private room,1,1.0,1.0,80.06,0,0,11,286,2,5.00,0.04,60,4804.0,No
3,6990,East Harlem,40.787780,-73.947590,Private room in rental unit,Private room,1,1.0,1.0,77.17,0,28,28,218,251,4.88,1.24,120,9260.0,Yes
4,7097,Fort Greene,40.691940,-73.973890,Private room in guest suite,Private room,2,1.0,2.0,202.47,0,0,0,106,423,4.89,2.12,255,51630.0,Yes


# 🏠 New York Airbnb SQL Analysis

## 📖 Objective

This notebook uses SQL to answer business questions related to Airbnb listings in New York. The analysis focuses on pricing, occupancy, revenue, customer ratings, host performance, and neighborhood trends to generate actionable business insights.

# 📊 Section 1: Market Overview

This section provides an overview of the New York Airbnb market, including the total number of listings, average pricing, estimated revenue, and the distribution of different accommodation types.

## 📌 Query 1: Total Airbnb Listings
## 🎯 Business Question

How many Airbnb listings are available in New York?

In [6]:
query = """
SELECT COUNT(*) AS total_listings
FROM airbnb_listings;
"""

pd.read_sql(query, conn)

,total_listings
0,30259


### 📈 Business Insight

The dataset contains 30,259 Airbnb listings, providing a large sample for analyzing pricing, revenue, occupancy, and customer behavior across New York.

## 📌 Query 2: Average Nightly Price

### 🎯Business Question
What is the average nightly price of an Airbnb listing?

In [7]:
query = """
SELECT ROUND(AVG(price),2) AS average_price
FROM airbnb_listings;
"""

pd.read_sql(query, conn)

,average_price
0,207.72


### 📈 Business Insight

The average nightly price serves as a benchmark for understanding the overall pricing level of Airbnb accommodations in New York and provides a baseline for comparing room types and neighborhoods.

## 📌 Query 3: Average Estimated Annual Revenue

### 🎯 Business Question

What is the average estimated annual revenue generated by Airbnb listings?

In [8]:
query = """
SELECT ROUND(AVG(estimated_revenue_l365d),2) AS average_revenue
FROM airbnb_listings;
"""

pd.read_sql(query, conn)

,average_revenue
0,12865.55


### 📈 Business Insight

The average estimated annual revenue highlights the overall earning potential of Airbnb listings and helps evaluate the profitability of the New York Airbnb market.

## 📌 Query 4: Distribution of Room Types

### 🎯 Business Question

Which room types are most commonly listed on Airbnb in New York?

In [9]:
query = """
SELECT room_type,
       COUNT(*) AS total_listings
FROM airbnb_listings
GROUP BY room_type
ORDER BY total_listings DESC;
"""

pd.read_sql(query, conn)

,room_type,total_listings
0,Entire home/apt,16808
1,Private room,12713
2,Hotel room,520
3,Shared room,218


### 📈 Business Insight

Entire homes/apartments represent the largest share of Airbnb listings, followed by private rooms, while hotel rooms and shared rooms account for a much smaller portion of the market.

## 📌 Query 5: Average Price by Room Type

### 🎯 Business Question

Which room type has the highest average nightly price?

In [10]:
query = """
SELECT room_type,
       ROUND(AVG(price),2) AS average_price
FROM airbnb_listings
GROUP BY room_type
ORDER BY average_price DESC;
"""

pd.read_sql(query, conn)

,room_type,average_price
0,Hotel room,435.95
1,Entire home/apt,251.94
2,Private room,141.51
3,Shared room,115.45


### 📈 Business Insight

Hotel rooms have the highest average nightly price, while shared rooms are the most affordable accommodation option. This reflects the differences in amenities, privacy, and guest experience offered by each room type.

## 📌 Query 6: Neighborhood Coverage

### 🎯 Business Question

How many unique neighborhoods are represented in the Airbnb dataset?

In [11]:
query = """
SELECT COUNT(DISTINCT neighbourhood_cleansed) AS total_neighbourhoods
FROM airbnb_listings;
"""

pd.read_sql(query, conn)

,total_neighbourhoods
0,223


### 📈 Business Insight

The dataset covers a wide range of neighborhoods across New York, allowing for comprehensive location-based pricing and revenue analysis.

## 📌 Query 7: Property Type Diversity

### 🎯 Business Question

How many different property types are listed on Airbnb?

In [12]:
query = """
SELECT COUNT(DISTINCT property_type) AS total_property_types
FROM airbnb_listings;
"""

pd.read_sql(query, conn)

,total_property_types
0,69


### 📈 Business Insight

The variety of property types demonstrates the diversity of accommodations available on Airbnb, ranging from private rooms to luxury homes and unique stays.

# 💰 Section 2: Pricing Analysis

This section analyzes Airbnb pricing patterns across different room types, neighborhoods, property types, and host categories. The goal is to identify factors that influence listing prices and understand market trends.

## 📌 Query 8: Top 10 Most Expensive Neighborhoods

### 🎯 Business Question

Which neighborhoods have the highest average Airbnb prices?

In [13]:
query = """
SELECT neighbourhood_cleansed,
       ROUND(AVG(price),2) AS average_price
FROM airbnb_listings
GROUP BY neighbourhood_cleansed
ORDER BY average_price DESC
LIMIT 10;
"""

pd.read_sql(query, conn)

,neighbourhood_cleansed,average_price
0,Hollis Hills,396.25
1,SoHo,350.40
2,Theater District,346.53
3,NoHo,343.14
4,DUMBO,340.90
5,Civic Center,338.30
6,Midtown,332.79
7,New Springville,328.17
8,Greenwich Village,322.76
9,Tribeca,298.08


### 📈 Business Insight

Luxury neighborhoods command significantly higher nightly prices than other areas. Understanding these premium markets helps identify high-value investment opportunities and pricing strategies.

## 📌 Query 9: Average Price by Property Type

### 🎯 Business Question

Which property types have the highest average nightly price?

In [14]:
query = """
SELECT property_type,
       ROUND(AVG(price),2) AS average_price
FROM airbnb_listings
GROUP BY property_type
ORDER BY average_price DESC
LIMIT 10;
"""

pd.read_sql(query, conn)

,property_type,average_price
0,Private room in resort,548.04
1,Room in serviced apartment,446.87
2,Houseboat,414.49
3,Entire serviced apartment,406.01
4,Room in bed and breakfast,399.23
5,Room in aparthotel,392.24
6,Room in boutique hotel,361.40
7,Boat,355.94
8,Room in hotel,323.77
9,Entire townhouse,295.72


### 📈 Business Insight

Luxury and unique accommodation types generally command higher prices, while standard rental units and private rooms remain more affordable for travelers.

## 📌 Query 10: Average Price by Bedrooms

### 🎯 Business Question

How does the average nightly price vary with the number of bedrooms?

In [15]:
query = """
SELECT bedrooms,
       ROUND(AVG(price),2) AS average_price
FROM airbnb_listings
GROUP BY bedrooms
ORDER BY bedrooms;
"""

pd.read_sql(query, conn)

,bedrooms,average_price
0,0.0,77.44
1,1.0,182.41
2,2.0,267.93
3,3.0,316.28
4,4.0,346.42
5,5.0,304.49
6,6.0,314.48
7,7.0,257.68
8,8.0,210.99
9,9.0,326.94


### 📈 Business Insight

Listings with more bedrooms generally command higher nightly prices, reflecting their ability to accommodate larger groups and offer additional space.

## 📌 Query 11: Superhost Pricing Comparison

### 🎯 Business Question

Do Superhosts charge higher prices than non-Superhosts?

In [16]:
query = """
SELECT host_is_superhost,
       ROUND(AVG(price),2) AS average_price
FROM airbnb_listings
GROUP BY host_is_superhost
ORDER BY average_price DESC;
"""

pd.read_sql(query, conn)

,host_is_superhost,average_price
0,Unknown,290.42
1,Yes,215.97
2,No,203.89


### 📈 Business Insight

Comparing average prices by Superhost status helps determine whether host reputation influences pricing strategies and guest willingness to pay.

## 📌 Query 12: Most Expensive Airbnb Listings

### 🎯 Business Question

Which Airbnb listings have the highest nightly prices?

In [17]:
query = """
SELECT id,
       neighbourhood_cleansed,
       property_type,
       room_type,
       price
FROM airbnb_listings
ORDER BY price DESC
LIMIT 10;
"""

pd.read_sql(query, conn)

,id,neighbourhood_cleansed,property_type,room_type,price
0,1547262972901712382,Hell's Kitchen,Entire rental unit,Entire home/apt,999.69
1,1456586441005764464,Upper East Side,Room in hotel,Private room,999.00
2,1122293005103628271,Greenwich Village,Entire rental unit,Entire home/apt,998.01
3,1653816332905227505,Financial District,Entire rental unit,Entire home/apt,998.00
4,1572779391857184852,Theater District,Entire rental unit,Entire home/apt,997.90
5,41371194,Midtown,Entire rental unit,Entire home/apt,997.14
6,33936370,Bushwick,Private room in rental unit,Private room,997.07
7,52706726,Inwood,Entire rental unit,Entire home/apt,995.57
8,591799481203483504,Wakefield,Entire vacation home,Entire home/apt,995.57
9,1679151620552800289,East Village,Entire rental unit,Entire home/apt,995.57


### 📈 Business Insight

Identifying the most expensive listings highlights the luxury segment of the Airbnb market and provides insight into premium pricing across different neighborhoods and property types.

# 🏡 Section 3: Property & Revenue Analysis

This section explores how different property types, room types, and neighborhoods contribute to Airbnb revenue and occupancy. The analysis helps identify the most profitable segments of the market.

## 📌 Query 13: Revenue by Room Type

### 🎯 Business Question

Which room type generates the highest average estimated annual revenue?

In [18]:
query = """
SELECT room_type,
       ROUND(AVG(estimated_revenue_l365d),2) AS average_revenue
FROM airbnb_listings
GROUP BY room_type
ORDER BY average_revenue DESC;
"""

pd.read_sql(query, conn)

,room_type,average_revenue
0,Shared room,16082.30
1,Entire home/apt,14919.78
2,Private room,10366.20
3,Hotel room,6222.10


### 📈 Business Insight

Comparing annual revenue across room types helps identify which accommodation category offers the highest earning potential for Airbnb hosts.

## 📌 Query 14: Revenue by Property Type

### 🎯 Business Question

Which property types generate the highest average estimated annual revenue?

In [19]:
query = """
SELECT property_type,
       ROUND(AVG(estimated_revenue_l365d),2) AS average_revenue
FROM airbnb_listings
GROUP BY property_type
ORDER BY average_revenue DESC
LIMIT 10;
"""

pd.read_sql(query, conn)

,property_type,average_revenue
0,Shared room in townhouse,101416.91
1,Room in aparthotel,75427.84
2,Private room in ranch,57714.00
3,Private room in cottage,49567.00
4,Shared room in serviced apartment,44505.00
5,Shared room in guesthouse,42671.00
6,Entire guest suite,40791.09
7,Private room in tiny home,36656.25
8,Private room in guest suite,34152.71
9,Entire bungalow,33290.89


### 📈 Business Insight

Some unique and luxury property types generate significantly higher annual revenue than standard accommodation options, highlighting premium market opportunities.

## 📌 Query 15: Most Profitable Neighborhoods

### 🎯 Business Question

Which neighborhoods generate the highest average estimated annual revenue?

In [20]:
query = """
SELECT neighbourhood_cleansed,
       ROUND(AVG(estimated_revenue_l365d),2) AS average_revenue
FROM airbnb_listings
GROUP BY neighbourhood_cleansed
ORDER BY average_revenue DESC
LIMIT 10;
"""

pd.read_sql(query, conn)

,neighbourhood_cleansed,average_revenue
0,DUMBO,39338.42
1,Fort Hamilton,35860.88
2,Middle Village,31792.23
3,Douglaston,28402.00
4,Eltingville,26275.67
5,Brownsville,25993.53
6,South Ozone Park,25645.86
7,Sea Gate,25365.44
8,Financial District,24297.78
9,Little Italy,23239.72


### 📈 Business Insight

High-revenue neighborhoods represent premium Airbnb markets where hosts can potentially achieve greater annual earnings.

## 📌 Query 16: Occupancy by Room Type

### 🎯 Business Question

Which room types have the highest average annual occupancy?

In [21]:
query = """
SELECT room_type,
       ROUND(AVG(estimated_occupancy_l365d),2) AS average_occupancy
FROM airbnb_listings
GROUP BY room_type
ORDER BY average_occupancy DESC;
"""

pd.read_sql(query, conn)

,room_type,average_occupancy
0,Private room,55.89
1,Shared room,51.40
2,Entire home/apt,51.00
3,Hotel room,14.53


### 📈 Business Insight

Comparing occupancy across room types helps identify which accommodations experience the highest booking demand throughout the year.

## 📌 Query 17: Revenue by Number of Bedrooms

### 🎯 Business Question

How does estimated annual revenue vary with the number of bedrooms?

In [22]:
query = """
SELECT bedrooms,
       ROUND(AVG(estimated_revenue_l365d),2) AS average_revenue
FROM airbnb_listings
GROUP BY bedrooms
ORDER BY bedrooms;
"""

pd.read_sql(query, conn)

,bedrooms,average_revenue
0,0.0,3929.80
1,1.0,9740.48
2,2.0,17446.38
3,3.0,28104.86
4,4.0,39860.36
5,5.0,53937.43
6,6.0,36334.21
7,7.0,4691.38
8,8.0,290867.38
9,9.0,8659.67


### 📈 Business Insight

Larger properties generally have the potential to generate higher annual revenue due to their greater accommodation capacity, although demand and pricing also influence earnings.

# ⭐ Section 4: Customer & Host Analysis

This section analyzes guest reviews, host performance, and Superhost status to understand customer satisfaction and the impact of host reputation on Airbnb listings.

## 📌 Query 18: Average Rating by Room Type

### 🎯 Business Question

Which room type receives the highest average guest rating?

In [23]:
query = """
SELECT room_type,
       ROUND(AVG(review_scores_rating),2) AS average_rating
FROM airbnb_listings
GROUP BY room_type
ORDER BY average_rating DESC;
"""

pd.read_sql(query, conn)

,room_type,average_rating
0,Entire home/apt,4.75
1,Shared room,4.73
2,Private room,4.71
3,Hotel room,4.39


### 📈 Business Insight

Comparing guest ratings across room types helps identify which accommodation categories consistently provide better guest experiences.

## 📌 Query 19: Superhost Performance

### 🎯 Business Question

Do Superhosts receive higher guest ratings than non-Superhosts?

In [24]:
query = """
SELECT host_is_superhost,
       ROUND(AVG(review_scores_rating),2) AS average_rating
FROM airbnb_listings
GROUP BY host_is_superhost
ORDER BY average_rating DESC;
"""

pd.read_sql(query, conn)

,host_is_superhost,average_rating
0,Yes,4.86
1,No,4.69
2,Unknown,4.54


### 📈 Business Insight

Superhost ratings provide insight into whether experienced and highly rated hosts consistently deliver better guest experiences.

## 📌 Query 20: Most Reviewed Neighborhoods

### 🎯 Business Question

Which neighborhoods have the highest average number of guest reviews?

In [25]:
query = """
SELECT neighbourhood_cleansed,
       ROUND(AVG(number_of_reviews),2) AS average_reviews
FROM airbnb_listings
GROUP BY neighbourhood_cleansed
ORDER BY average_reviews DESC
LIMIT 10;
"""

pd.read_sql(query, conn)

,neighbourhood_cleansed,average_reviews
0,Huguenot,206.33
1,New Dorp Beach,177.67
2,Clifton,95.73
3,Manhattan Beach,92.20
4,Concourse,86.05
5,Silver Lake,84.40
6,South Ozone Park,79.94
7,East Elmhurst,77.17
8,New Brighton,76.00
9,Grymes Hill,71.20


### 📈 Business Insight

Neighborhoods with higher review counts often indicate stronger guest demand and more established Airbnb activity.

## 📌 Query 21: Reviews by Room Type

### 🎯 Business Question

Which room types receive the highest average number of guest reviews?

In [26]:
query = """
SELECT room_type,
       ROUND(AVG(number_of_reviews),2) AS average_reviews
FROM airbnb_listings
GROUP BY room_type
ORDER BY average_reviews DESC;
"""

pd.read_sql(query, conn)

,room_type,average_reviews
0,Private room,40.40
1,Shared room,32.58
2,Entire home/apt,27.05
3,Hotel room,10.15


### 📈 Business Insight

Review volume helps identify accommodation types that attract more bookings and guest engagement.

## 📌 Query 22: Highest Rated Neighborhoods

### 🎯 Business Question

Which neighborhoods have the highest average guest ratings?

In [27]:
query = """
SELECT neighbourhood_cleansed,
       ROUND(AVG(review_scores_rating),2) AS average_rating
FROM airbnb_listings
GROUP BY neighbourhood_cleansed
HAVING COUNT(review_scores_rating) >= 10
ORDER BY average_rating DESC
LIMIT 10;
"""

pd.read_sql(query, conn)

,neighbourhood_cleansed,average_rating
0,Morris Park,4.93
1,Coney Island,4.93
2,City Island,4.92
3,Sea Gate,4.90
4,Randall Manor,4.90
5,Howard Beach,4.90
6,DUMBO,4.90
7,Red Hook,4.89
8,Cobble Hill,4.89
9,Belmont,4.89


### 📈 Business Insight

Filtering neighborhoods with at least 10 ratings helps ensure that the results are based on a meaningful number of guest reviews rather than a few exceptional cases.